### DISCUSSION TITLE
QMK firmware too large for crkbd/r2g

### DISCUSSION THREAD

**Discussion by sveeig** [manna-harbour/miryoku#76] [Q&A] [answered]
Using arch linux, I had some trouble flashing miryoku to this corne/r2g variant:

https://github.com/qmk/qmk_firmware/tree/master/keyboards/crkbd/r2g

After installing the qmk cli and doing `qmk setup` followed by `qmk flash -c -kb crkbd/r2g -km manna-harbour_miryoku`, The flash failed with the firmware beeing at more than 31k

---
 workaround

I did two things to make it fit:

1. acutally downgrade to avr-gcc 8.3.0 using `sudo downgrade avr-gcc`. That brought it down to 28.8k (still above the avail 28.6)
2. go into the $HOME/qmk_firmware folder that `qmk setup` downloaded, and edit `keyboards/crkbd/r2g/config.h`. Here I removed a couple of RGBLIGHT_EFFECT_* define's. I think one would have been enough but I took out four that I did not care for anyway. the relevant block after I removed christmas, knight, snake and rgb_test effects looks like this now:

![image](https://user-images.githubusercontent.com/674691/155549208-e30c255c-a633-45d4-96a1-4ccc1b7a2bcb.png)

**Comment by manna-harbour** [comment, marked as answer]
See https://github.com/qmk/qmk_firmware/blob/master/docs/squeezing_avr.md.

I plan on doing this automatically at some point, see https://github.com/manna-harbour/miryoku/issues/54.

**Comment by sveeig** [comment]
I just pulled the latest qmk master and latest miryoku branch/fork, merged them locally according to readme without workflow commits, miryoku now fits for crkbd/r2g with everything default. 

```
Checking file size of crkbd_r2g_manna-harbour_miryoku.hex                                           [WARNINGS]

 * The firmware size is approaching the maximum - 28128/28672 (98%, 544 bytes free)
```
